### PR TITLE
fix: nv21fix migration: correctly upgrade system actor

### PR DESCRIPTION
**Pull request by arajasek**
## Related Issues

The calibrationnet is currently in an inconsistent state, where there are miner actor CIDs that are not present in the system actor state. We need to backfix the `upgradeActorsV12Fix` to ALSO migrate the system actor state, this will leave us in a correct state.

## Proposed Changes

- Correctly migrate the system actor state at the v12 "fix"
- Ask users / miners to rollback history and reperform the migration.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

**Comment by arajasek** [comment]
TODO: 

- disable lotus migration caching
- confirm migration by comparing against "real" migrations in GST
- run on a devnet

**Review by raulk — APPROVED** [review]
Would be great if we could run a test in a local node against calibrationnet, but running the upgrade in a devnet would be enough. We won't get to test the rollback and redo mechanics that will occur in Calibnet, but folks have done this operation before and the code hasn't changed, so from the risk/gain perspective, doesn't make me too uncomfortable.

File: miner/miner.go [hunk 1]
```diff
@@ -324,7 +324,7 @@ minerLoop:
 					"block-time", btime, "time", build.Clock.Now(), "difference", build.Clock.Since(btime))
 			}
 
-			if os.Getenv("LOTUS_MINER_NO_SLASHFILTER") != "_yes_i_know_i_can_and_probably_will_lose_all_my_fil_and_power_" {
+			if os.Getenv("LOTUS_MINER_NO_SLASHFILTER") != "_yes_i_know_i_can_and_probably_will_lose_all_my_fil_and_power_" && !build.IsNearUpgrade(base.TipSet.Height(), build.UpgradeWatermelonFixHeight) {
```
Comment (Stebalien) [inline miner/miner.go:327]: ditto

File: chain/vm/fvm.go [hunk 2]
```diff
@@ -92,6 +92,15 @@ func (x *FvmExtern) VerifyConsensusFault(ctx context.Context, a, b, extra []byte
 		log.Info("invalid consensus fault: submitted blocks are the same")
 		return ret, totalGas
 	}
+
+	// workaround chain halt
```
Comment (LesnyRumcajs) [inline chain/vm/fvm.go:96]: This will also apply on other networks, not only calibnet, right?

**Review by Stebalien — COMMENTED** [review]
Not blocking a merge.

**Review by Stebalien — APPROVED** [review]

**Comment by LesnyRumcajs** [comment]
@arajasek will the check 
```go
		if inActor.Head != outActor.Head {
			return xerrors.Errorf("mismatched head for actor %s", a)
		}
```
still hold?

**Comment by arajasek** [comment]
> @arajasek will the check
> 
> ```go
> 		if inActor.Head != outActor.Head {
> 			return xerrors.Errorf("mismatched head for actor %s", a)
> 		}
> ```
> 
> still hold?

I've added an exception for it, thank you!

**Pull request merged**